qa/packaging/src/test/java/org/elasticsearch/packaging/test/DebMetadataTests.java

```diff
@@ -35,15 +48,26 @@ public void test05CheckLintian() {
         if (helpText.contains("--fail-on-warnings")) {
             extraArgs = "--fail-on-warnings";
         } else if (helpText.contains("--fail-on error")) {
-            extraArgs = "--fail-on warning";
-            // Recent lintian versions are picky about malformed or mismatched overrides.
-            // Unfortunately override syntax changes between lintian versions in a non-backwards compatible
-            // way, so we have to tolerate these (or maintain separate override files per lintian version).
-            if (helpText.contains("--suppress-tags")) {
-                extraArgs += " --suppress-tags malformed-override,mismatched-override";
+            extraArgs = "--fail-on error,warning";
```

Comment:
Lintian `--fail-on` requires all levels, to fail on, to be listed .
> 'warning' does not imply 'error'
https://manpages.debian.org/testing/lintian/lintian.1.en.html